src/App.tsx

```diff
@@ -1,74 +1,153 @@
+type State = {
+  query: string;
+  selectedPerson: Person | null;
+  isDropedDown: boolean;
+};
+
+export class App extends React.Component {
+  state: State = {
+    query: '',
+    selectedPerson: null,
+    isDropedDown: false,
+  };
+
+  getFilterPeople = (): Person[] => {
+    const { query } = this.state;
+
+    if (!query) {
+      return peopleFromServer;
+    }
+
+    return peopleFromServer.filter(person =>
+      person.name.toLowerCase().includes(query.toLowerCase().trim()),
+    );
+  };
+
+  handlerBlur = () => {
+    setTimeout(() => {
+      this.setState({ isDropedDown: false });
+    }, 300);
+  };
+
+  handlerOnChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = e.target;
 
-          <div className="dropdown-menu" role="menu" data-cy="suggestions-list">
-            <div className="dropdown-content">
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Haverbeke</p>
-              </div>
+    this.setState({ query: value });
 
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Bernard Haverbeke</p>
-              </div>
+    if (!value) {
+      this.setState({ selectedPerson: null });
+    }
+  };
 
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Antone Haverbeke</p>
-              </div>
+  handlerClear = () => {
+    this.setState({
+      selectedPerson: null,
+      query: '',
+    });
+  };
 
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-danger">Elisabeth Haverbeke</p>
-              </div>
+  render() {
+    const { query, selectedPerson } = this.state;
+    const { name, born, died } = selectedPerson || {};
 
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter de Decker</p>
-              </div>
+    return (
+      <div className="container">
+        <main className="section is-flex is-flex-direction-column">
+          {name ? (
+            <h1 className="title" data-cy="title">
+              {`${name} (${born} - ${died})`}
+            </h1>
+          ) : (
+            <h1 className="title" data-cy="title">
+              No selected person
+            </h1>
+          )}
 
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-danger">Petronella de Decker</p>
-              </div>
+          <div className="dropdown is-active">
+            <div className="dropdown-trigger">
+              <input
+                type="text"
+                placeholder="Enter a part of the name"
+                className="input"
+                data-cy="search-input"
+                onFocus={() => this.setState({ isDropedDown: true })}
+                onBlur={this.handlerBlur}
```

Comment:
Use [destruction assignment ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) to `this` and `this.state`